iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/DataRegion.java

```diff
@@ -885,16 +885,14 @@ public void insert(InsertRowNode insertRowNode) throws WriteProcessException {
               && insertRowNode.getTime()
                   > lastFlushTimeMap.getFlushedTime(timePartitionId, insertRowNode.getDeviceID());
 
-      Map<TsFileProcessor, Boolean> tsFileProcessorMapForFlushing = new HashMap<>();
-
       // insert to sequence or unSequence file
-      insertToTsFileProcessor(
-          insertRowNode, isSequence, timePartitionId, tsFileProcessorMapForFlushing);
+      TsFileProcessor tsFileProcessor =
+          insertToTsFileProcessor(insertRowNode, isSequence, timePartitionId);
 
       // check memtable size and may asyncTryToFlush the work memtable
-      for (Map.Entry<TsFileProcessor, Boolean> entry : tsFileProcessorMapForFlushing.entrySet()) {
-        if (entry.getKey().shouldFlush()) {
-          fileFlushPolicy.apply(this, entry.getKey(), entry.getValue());
+      if (tsFileProcessor != null) {
+        if (tsFileProcessor.shouldFlush()) {
```

Comment:
Maybe `if (tsFileProcessor != null && tsFileProcessor.shouldFlush())` is better.

Reply:
fixed